apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -659,7 +659,40 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
           ttl: db_max_ttl
         )
 
-      :ok = insert_result
+      case insert_result do
+        {:error, :unset_not_allowed} ->
+          Logger.warning("Tried to unset a property with `allow_unset`=false.",
+            tag: "unset_not_allowed"
+          )
+
+          MessageTracker.discard(new_state.message_tracker, message_id)
+
+          :telemetry.execute(
+            [:astarte, :data_updater_plant, :data_updater, :discarded_message],
+            %{},
+            %{realm: new_state.realm}
+          )
+
+          base64_payload = Base.encode64(payload)
+
+          error_metadata = %{
+            "interface" => inspect(interface),
+            "path" => inspect(path),
+            "base64_payload" => base64_payload
+          }
+
+          execute_device_error_triggers(
+            new_state,
+            "unset_not_allowed",
+            error_metadata,
+            timestamp
+          )
+
+          raise "Forbidden"
```

Comment:
Once we've handled the message, sent the error trigger and updated the telemetry, there's no need to crash the DataUpdater process too. After all, DUP does not expose an API

Reply:
:+1: from what I understand the execution should stop there, right? Otherwise an ack is sent and telemetry is updated for the correct processing of the message.